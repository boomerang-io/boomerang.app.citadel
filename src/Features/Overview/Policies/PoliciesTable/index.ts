// @ts-expect-error ts-migrate(6142) FIXME: Module './PoliciesTable' was resolved to 'C:/Users... Remove this comment to see the full error message
export { default } from "./PoliciesTable"