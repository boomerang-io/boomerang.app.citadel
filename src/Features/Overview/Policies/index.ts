// @ts-expect-error ts-migrate(6142) FIXME: Module './Policies' was resolved to 'C:/Users/Isab... Remove this comment to see the full error message
export { default } from "./Policies"