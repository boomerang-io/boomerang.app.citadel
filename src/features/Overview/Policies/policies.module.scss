.container {
  display: flex;
  flex-direction: column;
  margin-top: 2rem;
  position: relative;
}

.title {
  padding: 0;
}

.message {
  color: #242a2e;
  font-size: 0.875rem;
  letter-spacing: 0.0625rem;
  line-height: 0.92rem;
  text-align: left;
  text-overflow: ellipsis;
  word-break: break-word;
}
.button {
  position: absolute;
  right: 0rem;
  top: 1rem;
}
