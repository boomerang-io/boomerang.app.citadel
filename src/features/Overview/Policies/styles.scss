@import "~@boomerang/boomerang-styles/lib/colors";

.c-policies {
  display: flex;
  flex-direction: column;
  margin-top: 2rem;
  position: relative;
}

.b-policies-header__title {
  padding: 1rem 0;
}

.b-policies-header__message {
  font-weight: 400;
  line-height: 1.5rem;
}
.b-policies-header__button {
  position: absolute;
  right: 0rem;
  top: 1rem;
}
