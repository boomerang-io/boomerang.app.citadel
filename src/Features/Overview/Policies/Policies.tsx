import React from "react";
// @ts-expect-error ts-migrate(2307) FIXME: Cannot find module 'Components/NoDisplay' or its c... Remove this comment to see the full error message
import NoDisplay from "Components/NoDisplay";
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module '@boo... Remove this comment to see the full error message
import { Button } from "@boomerang-io/carbon-addons-boomerang-react";
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module 'reac... Remove this comment to see the full error message
import { useHistory } from "react-router-dom";
import PoliciesTable from "./PoliciesTable";
// @ts-expect-error ts-migrate(2307) FIXME: Cannot find module 'Config/appConfig' or its corre... Remove this comment to see the full error message
import { appLink } from "Config/appConfig";
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module '@car... Remove this comment to see the full error message
import { Add16 } from "@carbon/icons-react";
// @ts-expect-error ts-migrate(2307) FIXME: Cannot find module './policies.module.scss' or its... Remove this comment to see the full error message
import styles from "./policies.module.scss";

type Props = {
    policies?: any[];
    activeTeamId?: string;
};

export function Policies({ policies, activeTeamId }: Props) {
  let history = useHistory();
  return (
    // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
    <section className={styles.container} data-testid="policies-container" id="policies-container">
      {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
      <h2 className={styles.title}>{`Policies (${policies.length})`}</h2>
      {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
      <p className={styles.message}>
        All of your teams policies can be found here. Go forth and define rules to enforce your standards!
      </p>
      {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
      <div className={styles.button}>
        {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
        <Button
          data-testid="add-policy-button"
          iconDescription="Create Policy"
          onClick={() => history.push(appLink.createPolicy({teamId: activeTeamId}))}
          renderIcon={Add16}
          size="field"
        >
          Create Policy
        </Button>
      </div>
      {/* @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'. */}
      {policies.length > 0 ? (
        // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
        <PoliciesTable policies={policies} />
      ) : (
        // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
        <NoDisplay text="No policies found. Go forth and create one or two why don't you?" style={{ width: "20rem" }} />
      )}
    </section>
  );
}

export default Policies;
