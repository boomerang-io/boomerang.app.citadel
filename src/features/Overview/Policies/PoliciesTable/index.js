import React, { Component } from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import { DataTable } from "carbon-components-react";
import styles from "./policiesTable.module.scss";
import "./styles.scss";

export class PoliciesTable extends Component {
  static propTypes = {
    poicies: PropTypes.array,
    definitions: PropTypes.array
  };
  headers = [
    {
      header: "Name",
      key: "name"
    },

    {
      header: "Definitions",
      key: "definitions"
    },
    {
      header: "Rules",
      key: "definitions"
    },
    {
      header: "Stage Gate Allocations",
      key: "stages"
    }
  ];

  handleRowClick = row => {
    const { location, history } = this.props;
    history.push(`${location.pathname}/policy/edit/${row.id}`);
  };

  renderCell = (rowIndex, cellIndex, value) => {
    const column = this.headers[cellIndex];
    switch (column.header) {
      case "Name":
        return <p className={styles.tableTextarea}>{value}</p>;
      case "Definitions":
        return <p className={styles.tableTextarea}>{Array.isArray(value) ? value.length : "---"}</p>;
      case "Rules":
        return (
          <p className={styles.tableTextarea}>
            {Array.isArray(value)
              ? value.reduce((accum, definition) => {
                  accum += definition.rules.length;
                  return accum;
                }, 0)
              : "---"}
          </p>
        );

      case "Stage Gate Allocations":
        return <p className={styles.tableTextarea}>{!!value.length ? value.join(", ") : "---"}</p>;
      case "":
        return (
          <div className={styles.tableIcons}>
            {/* <div>
              <Icon
                data-tip
                data-for={`policies-table-${value}-${cellIndex}-edit`}
                className={styles.tableIcon}
                name="icon--edit"
                alt="Edit Policy"
              />
              <Tooltip id={`policies-table-${value}-${cellIndex}-edit`}>Edit</Tooltip>
            </div>
            <div>
              <Icon
                data-tip
                data-for={`policies-table-${value}-${cellIndex}-delete`}
                className={styles.tableIcon}
                name="icon--delete"
                alt="Delete Policy"
              />
              <Tooltip id={`policies-table-${value}-${cellIndex}-delete`}>Delete</Tooltip>
            </div> */}
          </div>
        );
      default:
        return value || "---";
    }
  };

  render() {
    const { policies } = this.props;
    const { TableContainer, Table, TableHead, TableRow, TableBody, TableCell, TableHeader } = DataTable;
    return (
      <DataTable
        rows={policies}
        headers={this.headers}
        render={({ rows, headers, getHeaderProps }) => (
          <TableContainer>
            <Table className={styles.tableContainer} sortable={true} zebra={false}>
              <TableHead>
                <TableRow className={styles.tableHeadRow}>
                  {headers.map(header => (
                    <TableHeader {...getHeaderProps({ header, className: `${styles.tableHeader}` })}>
                      {header.header}
                    </TableHeader>
                  ))}
                </TableRow>
              </TableHead>
              <TableBody className={styles.tableBody} data-testid="policies-tbody">
                {rows.map((row, rowIndex) => (
                  <TableRow key={row.id} className={styles.tableRow} onClick={() => this.handleRowClick(row)}>
                    {row.cells.map((cell, cellIndex) => (
                      <TableCell key={`${cell.id}-${cellIndex}`} style={{ padding: "0" }}>
                        <div className={styles.tableCell}>{this.renderCell(rowIndex, cellIndex, cell.value)}</div>
                      </TableCell>
                    ))}
                  </TableRow>
                ))}
              </TableBody>
            </Table>
          </TableContainer>
        )}
      />
    );
  }
}

export default withRouter(PoliciesTable);
