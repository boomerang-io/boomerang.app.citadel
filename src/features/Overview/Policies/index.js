import React from "react";
import PropTypes from "prop-types";
import NoDisplay from "components/NoDisplay";
import { Button } from "carbon-components-react";
import { useHistory, useLocation } from "react-router-dom";
import PoliciesTable from "./PoliciesTable";
import { Add16 } from "@carbon/icons-react";
import styles from "./policies.module.scss";

Policies.propTypes = {
  policies: PropTypes.array
};

export function Policies({ policies }) {
  let history = useHistory();
  let location = useLocation();
  return (
    <section className={styles.container} data-testid="policies-container" id="policies-container">
      <h2 className={styles.title}>{`Policies (${policies.length})`}</h2>
      <p className={styles.message}>
        All of your teams policies can be found here. Go forth and define rules to enforce your standards!
      </p>
      <div className={styles.button}>
        <Button
          data-testid="add-policy-button"
          iconDescription="Add Policy"
          onClick={() => history.push(`${location.pathname}/policy/create`)}
          renderIcon={Add16}
          size="field"
        >
          Add Policy
        </Button>
      </div>
      {policies.length > 0 ? (
        <PoliciesTable policies={policies} />
      ) : (
        <NoDisplay message="No policies to be found. Add some!" style={{ width: "20rem" }} />
      )}
    </section>
  );
}

export default Policies;
