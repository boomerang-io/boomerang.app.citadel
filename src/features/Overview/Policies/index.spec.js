import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import Policies from "./index";

const policies = [
  {
    "createdDate": 1558051200000,
    "name": "Timss",
    "teamId": "5a8b331e262a70306622df73",
    "definitions": [
      {
        "ciPolicyDefinitionId": "5cd49777f6ea74a9bb6ac629",
        "rules": [
          {
            "value": "sdfgsdfg",
            "operator": "less than",
            "metric": "complexity"
          }
        ]
      },
      {
        "ciPolicyDefinitionId": "5cd498f3f6ea74a9bb6ad0f3",
        "rules": [
          {
            "version": " asdfasdf",
            "artifact": "asdasdf`",
            "type": "maven"
          }
        ]
      }
    ],
    "stages":[
      "dev"
   ],
    "id": "R4nuqJE"
  }
];

describe("Policies --- Snapshot", () => {
  it("Capturing Snapshot of Policies", () => {
    const renderedValue = renderer
      .create(
        <MemoryRouter>
          <Policies
            policies={policies}
          />
        </MemoryRouter>
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("Policies --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <MemoryRouter>
        <Policies
          policies={policies}
        />
      </MemoryRouter>
    );
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
