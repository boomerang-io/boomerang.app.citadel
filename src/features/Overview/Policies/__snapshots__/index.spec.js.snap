// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Policies --- Snapshot Capturing Snapshot of Policies 1`] = `
<div
  className="container"
  data-testid="policies-container"
  id="policies-container"
>
  <div>
    <h2
      className="title"
    >
      Policies (1)
    </h2>
    <p
      className="message"
    >
      All of your teams policies can be found here. Go forth and define rules to enforce your standards!
    </p>
    <div
      className="button"
    >
      <button
        className="bx--btn bx--btn--field bx--btn--primary"
        data-testid="add-policy-button"
        disabled={false}
        onClick={[Function]}
        tabIndex={0}
        type="button"
      >
        Add Policy
        <svg
          aria-hidden="true"
          aria-label="Add Policy"
          className="bx--btn__icon"
          focusable="false"
          height={16}
          preserveAspectRatio="xMidYMid meet"
          role="img"
          style={
            Object {
              "willChange": "transform",
            }
          }
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 7V3H7v4H3v2h4v4h2V9h4V7z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="bx--data-table-container"
  >
    <table
      className="bx--data-table tableContainer bx--data-table--no-border"
      sortable="true"
    >
      <thead>
        <tr
          className="tableHeadRow"
        >
          <th
            className="tableHeader"
            scope="col"
          >
            <span
              className="bx--table-header-label"
            >
              Name
            </span>
          </th>
          <th
            className="tableHeader"
            scope="col"
          >
            <span
              className="bx--table-header-label"
            >
              Definitions
            </span>
          </th>
          <th
            className="tableHeader"
            scope="col"
          >
            <span
              className="bx--table-header-label"
            >
              Rules
            </span>
          </th>
          <th
            className="tableHeader"
            scope="col"
          >
            <span
              className="bx--table-header-label"
            >
              Stage Gate Allocations
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        aria-live="polite"
        className="tableBody"
        data-testid="policies-tbody"
      >
        <tr
          className="tableRow"
          onClick={[Function]}
        >
          <td
            style={
              Object {
                "padding": "0",
              }
            }
          >
            <div
              className="tableCell"
            >
              <p
                className="tableTextarea"
              >
                Timss
              </p>
            </div>
          </td>
          <td
            style={
              Object {
                "padding": "0",
              }
            }
          >
            <div
              className="tableCell"
            >
              <p
                className="tableTextarea"
              >
                2
              </p>
            </div>
          </td>
          <td
            style={
              Object {
                "padding": "0",
              }
            }
          >
            <div
              className="tableCell"
            >
              <p
                className="tableTextarea"
              >
                2
              </p>
            </div>
          </td>
          <td
            style={
              Object {
                "padding": "0",
              }
            }
          >
            <div
              className="tableCell"
            >
              <p
                className="tableTextarea"
              >
                dev
              </p>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
